### PR TITLE
Make output directory a build input

### LINKED ISSUE #41861: flutter build bundle only one bundle can exist at a time
Split of https://github.com/flutter/flutter/issues/41352#issuecomment-537709454
I make heavy uses of `build bundle` and have found out an issue that appears on `Flutter 1.10.7-pre.123 ` and leads to the "could not resolve the kernel binary" error.
This issue appears when using the `--asset-dir` flag. 

flutter version:
```log
Flutter 1.10.7-pre.123 • channel master • http://github.com/flutter/flutter
Framework • revision bedf46d06e (6 minutes ago) • 2019-10-02 15:21:45 -0700
Engine • revision aa8fcbb652
Tools • Dart 2.6.0 (build 2.6.0-dev.4.0 98b286de21)
``` 

Steps to reproduce:
```sh
$ flutter build bundle --asset-dir /tmp/test
$ tree /tmp/test # looks good
/tmp/test
├── AssetManifest.json
├── FontManifest.json
├── isolate_snapshot_data
├── kernel_blob.bin
├── LICENSE
├── packages
│   └── cupertino_icons
│       └── assets
│           └── CupertinoIcons.ttf
└── vm_snapshot_data
$ flutter build bundle --asset-dir /tmp/test2
$ tree /tmp/test2 # also looks good
/tmp/test2
├── AssetManifest.json
├── FontManifest.json
├── isolate_snapshot_data
├── kernel_blob.bin
├── LICENSE
├── packages
│   └── cupertino_icons
│       └── assets
│           └── CupertinoIcons.ttf
└── vm_snapshot_data
# BUT
$ tree /tmp/test # gets deleted
/tmp/test
└── packages
    └── cupertino_icons
        └── assets
```
Pinging @jonahwilliams as I think it is related to tooling.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This allows repeated invocations of build bundle to not delete old invocations with the same arguments besides build directory.

Fixes https://github.com/flutter/flutter/issues/41861

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/build_system.dart [hunk 1]
```diff
@@ -294,6 +294,7 @@ class Environment {
     }
     // in case there was no configuration, provide some value.
     buffer.write('Flutter is awesome');
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/build_system/build_system.dart]: Do you need this awesome placeholder anymore?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/build_system.dart]: Done!

Comment (jmagman) [reply packages/flutter_tools/lib/src/build_system/build_system.dart]: 😢

**Review by jmagman — APPROVED** [review]
LGTM

**Review by stuartmorgan-g — APPROVED** [review]

**Pull request merged**